linux/keyman-config/keyman_config/gnome_keyboards_util.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/python3
+import logging
+import os
+from gi.repository import Gio
+
+from gi.overrides.GLib import Variant
+
+
+class GnomeKeyboardsUtil():
+    def __init__(self):
+        self.input_sources = Gio.Settings.new("org.gnome.desktop.input-sources")
+
+    def read_input_sources(self):
+        sourcesVal = self.input_sources.get_value("sources")
+        sources = self._convert_variant_to_array(sourcesVal)
+        return sources
+
+    def write_input_sources(self, sources):
+        sourcesVal = self._convert_array_to_variant(sources)
+        self.input_sources.set_value("sources", sourcesVal)
+
+    def _convert_variant_to_array(self, variant):
+        if variant is None:
+            return []
+
+        values = []
+        # a(ss)
+        nChildren = variant.n_children()
+        for i in range(nChildren):
+            # (ss)
```

Comment:
I don't understand this comment either :)

Reply:
done